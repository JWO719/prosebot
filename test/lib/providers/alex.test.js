const Alex = require('../../../lib/providers/alex')

describe('Alex provider', () => {
  let provider

  beforeEach(() => {
    const map = new Map()

    const obj = {
      'guys.md': 'Hey guys how are you',
      'he.md': 'He walked to class.'
    }

    for (const key in obj) {
      map.set(key, obj[key])
    }

    provider = new Alex(map)
  })

  describe('#buildResults', () => {
    it('returns the expected result', () => {
      const actual = provider.buildResults()
      expect(actual).toMatchSnapshot()
    })
  })
})
