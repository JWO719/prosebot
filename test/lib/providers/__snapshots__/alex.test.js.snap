// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alex provider #buildResults returns the expected result 1`] = `
Map {
  "guys.md" => Array [
    Object {
      "annotation_level": "warning",
      "line": 1,
      "reason": "\`guys\` may be insensitive, use \`people\`, \`persons\`, \`folks\` instead",
    },
  ],
  "he.md" => Array [
    Object {
      "annotation_level": "warning",
      "line": 1,
      "reason": "\`He\` may be insensitive, use \`They\`, \`It\` instead",
    },
  ],
}
`;
